addons/base_automation/models/base_automation.py

```diff
@@ -741,10 +742,17 @@ def write(self, vals, **kw):
                 # check preconditions on records
                 pre = {a: a._filter_pre(records) for a in automations}
                 # read old values before the update
-                old_values = {
-                    record.id: {field_name: record[field_name] for field_name in vals}
-                    for record in records
-                }
+                old_values = {}
+                for record in records:
+                    for fname in vals:
+                        fname_val = False
+                        if is_selection_groups(fname):
+                            old_group_id = (record.groups_id.filtered(lambda g: g.id in get_selection_groups(fname)) - record.groups_id.implied_ids).ids
+                            if len(old_group_id) == 1:
+                                fname_val = old_group_id.pop()
+                        else:
+                            fname_val = record[fname]
+                        old_values[record.id] = {fname: fname_val}
```

Comment:
This is not correct. it just keeps the last value set.

Could you just add an `if` to the comprehension that was present?
Maybe you can just ignore these selection group fields instead of trying to map them to groups?
FYI, we will probably remove selection groups somewhere this year.

Reply:
I am unsure exactly what you mean? The old_values is always meant to keep the last value set in order to use it as context for the automation, this PR is trying to ensure that the correct old_values are being populated, and that they are being created in a way to avoid errors going forward. If you could help clarify my misunderstandings I would appreciate it!

I could definitely add a check to avoid this type of field entirely in the list comprehension, but I am not sure if that would maintain the same intended functionality of this code snippet, since the old_values would not hold the information about what groups the user was a part of prior to the write() call being executed.

Thanks for taking a look!

Reply:
If `len(vals) > 1`, you will get only one value per record in the `old_values` dict.

These selection fields are quite special (that's why we want to remove them in the future). They are mapped to changes on groups, I don't think they are functional in this context today - that's why I suggest to simply ignore them.

Reply:
I see the confusion, the `len(vals) == 1` condition only executes on the output of the set subtraction between the filtered `groups_id` and the `implied_ids` which _should_ only ever yield one value. The condition is a failsafe in case somehow multiple values are yielded, as a single user cannot be assigned as being multiple group roles for a single selection field. Hopefully this makes more sense? I may have still messed up the logic somewhere, but at least that was the intention.

That being said, I definitely can see that these fields likely wouldn't be relevant in this context, and I will amend the PR accordingly. I appreciate the explanation!